app/views/layouts/application.html.erb

```diff
@@ -28,6 +28,7 @@
           </ul>
           <ul class="nav navbar-nav navbar-right">
           <% if user_signed_in? %>
+            <li><%= link_to 'My Profile', user_path(current_user) %></li>
```

Comment:
It looks like the user_id is passed through the URL, but the value of it is disregarded and `current_user` is used instead.  I think we should either not pass through the user_id in the URL or we should show the dashboard for the user that is passed along.  Probably the second.